tests/discrete_domain.cpp

```diff
@@ -234,3 +237,80 @@ TEST(ProductMDomainTest, SliceDomainXToolate)
             R"rgx([Aa]ssert.*uid<ODDims>\(m_element_end\).*uid<ODDims>\(odomain\.m_element_end\).*)rgx");
 #endif
 }
+
+TEST(ProductMDomainTest, Transpose2DConstructor)
+{
+    DDomXY const dom_x_y(lbound_x_y, nelems_x_y);
+    DDomYX const dom_y_x(dom_x_y);
+    EXPECT_EQ(ddc::select<DDimX>(dom_y_x.front()), ddc::select<DDimX>(lbound_x_y));
+    EXPECT_EQ(ddc::select<DDimY>(dom_y_x.front()), ddc::select<DDimY>(lbound_x_y));
+    EXPECT_EQ(ddc::select<DDimX>(dom_y_x.back()), ddc::select<DDimX>(ubound_x_y));
+    EXPECT_EQ(ddc::select<DDimY>(dom_y_x.back()), ddc::select<DDimY>(ubound_x_y));
+}
+
+TEST(ProductMDomainTest, Transpose3DConstructor)
+{
+    DDomX const dom_x(lbound_x, nelems_x);
+    DDomY const dom_y(lbound_y, nelems_y);
+    DDomZ const dom_z(lbound_z, nelems_z);
+    DDomXYZ const dom_x_y_z(dom_x, dom_y, dom_z);
+    DDomZYX const dom_z_y_x(dom_x_y_z);
+    EXPECT_EQ(ddc::select<DDimX>(dom_x_y_z.front()), ddc::select<DDimX>(dom_z_y_x.front()));
+    EXPECT_EQ(ddc::select<DDimY>(dom_x_y_z.front()), ddc::select<DDimY>(dom_z_y_x.front()));
+    EXPECT_EQ(ddc::select<DDimZ>(dom_x_y_z.front()), ddc::select<DDimZ>(dom_z_y_x.front()));
+    EXPECT_EQ(ddc::select<DDimX>(dom_x_y_z.back()), ddc::select<DDimX>(dom_z_y_x.back()));
+    EXPECT_EQ(ddc::select<DDimY>(dom_x_y_z.back()), ddc::select<DDimY>(dom_z_y_x.back()));
+    EXPECT_EQ(ddc::select<DDimZ>(dom_x_y_z.back()), ddc::select<DDimZ>(dom_z_y_x.back()));
+}
+
+TEST(ProductMDomainTest, Transpose2DAssignConstructor)
+{
+    DDomXY const dom_x_y(lbound_x_y, nelems_x_y);
+    DDomYX const dom_y_x = dom_x_y;
+    EXPECT_EQ(ddc::select<DDimX>(dom_y_x.front()), ddc::select<DDimX>(lbound_x_y));
+    EXPECT_EQ(ddc::select<DDimY>(dom_y_x.front()), ddc::select<DDimY>(lbound_x_y));
+    EXPECT_EQ(ddc::select<DDimX>(dom_y_x.back()), ddc::select<DDimX>(ubound_x_y));
+    EXPECT_EQ(ddc::select<DDimY>(dom_y_x.back()), ddc::select<DDimY>(ubound_x_y));
+}
+
+TEST(ProductMDomainTest, Transpose3DAssignConstructor)
+{
+    DDomX const dom_x(lbound_x, nelems_x);
+    DDomY const dom_y(lbound_y, nelems_y);
+    DDomZ const dom_z(lbound_z, nelems_z);
+    DDomXYZ const dom_x_y_z(dom_x, dom_y, dom_z);
+    DDomZYX const dom_z_y_x = dom_x_y_z;
+    EXPECT_EQ(ddc::select<DDimX>(dom_x_y_z.front()), ddc::select<DDimX>(dom_z_y_x.front()));
+    EXPECT_EQ(ddc::select<DDimY>(dom_x_y_z.front()), ddc::select<DDimY>(dom_z_y_x.front()));
+    EXPECT_EQ(ddc::select<DDimZ>(dom_x_y_z.front()), ddc::select<DDimZ>(dom_z_y_x.front()));
+    EXPECT_EQ(ddc::select<DDimX>(dom_x_y_z.back()), ddc::select<DDimX>(dom_z_y_x.back()));
+    EXPECT_EQ(ddc::select<DDimY>(dom_x_y_z.back()), ddc::select<DDimY>(dom_z_y_x.back()));
+    EXPECT_EQ(ddc::select<DDimZ>(dom_x_y_z.back()), ddc::select<DDimZ>(dom_z_y_x.back()));
+}
```

Comment:
@tpadioleau These 2 tests don't compile however this is because the transpose constructor is marked `explicit`. I think this is something that should be kept so these tests should be removed. Do you agree?

Reply:
Should work with `DDomZYX const dom_z_y_x(dom_x_y_z)`

Reply:
Yes. That is the test that I added above

Reply:
Sorry I did not see the other tests. Yes I would remove them.

Reply:
Done [6ff8339](https://github.com/CExA-project/ddc/pull/420/commits/6ff8339308a8f2bda1c85e65a13695276cb805ee)